WooCommerce/Classes/ServiceLocator/ProductImageUploader.swift

```diff
@@ -169,7 +169,16 @@ final class ProductImageUploader: ProductImageUploaderProtocol {
             imagesSaver = ProductImagesSaver(siteID: siteID, productID: productID, stores: stores)
             imagesSaverByProduct[key] = imagesSaver
         }
-        imagesSaver.saveProductImagesWhenNoneIsPendingUploadAnymore(imageActionHandler: handler, onProductSave: onProductSave)
+        imagesSaver.saveProductImagesWhenNoneIsPendingUploadAnymore(imageActionHandler: handler) { [weak self] result in
+            guard let self = self else { return }
+            if case let .failure(error) = result {
+                self.errorsSubject.send(.init(siteID: siteID,
+                                              productID: productID,
+                                              productImageStatuses: handler.productImageStatuses,
+                                              error: .savingProductImages(error: error)))
+            }
+            onProductSave(result)
```

Comment:
Super nit:
Would it make sense to call the `onProductSave` completion handler first before sending value through `errorsSubject`?

I am just thinking of an imaginary scenario in which a loading indicator can be dismissed by the view controller before displaying the error received by `errorsSubject`.

Reply:
Yea that's a good point! Updated in https://github.com/woocommerce/woocommerce-ios/pull/7154/commits/5e4da680ae5881d818e38403e77b61c578f05847